server/src/main/java/io/crate/types/StorageSupport.java

```diff
@@ -61,6 +66,26 @@ public abstract ValueIndexer<? super T> valueIndexer(
         Reference ref,
         Function<ColumnIdent, Reference> getRef);
 
+    /**
+     * Decode a value from bytes in a stored field
+     */
+    public Object decodeFromBytes(ColumnIdent column, SourceParser sourceParser, byte[] bytes) {
+        throw new UnsupportedOperationException("decodeFromBytes not supported");
+    }
+
+    /**
+     * Decode a value from a long in a stored field
+     */
+    public Object decodeFromLong(long input) {
+        throw new UnsupportedOperationException("decodeFromLong not supported");
+    }
+
+    /**
+     * Decode a value from an int in a stored field
+     */
+    public Object decodeFromInt(int input) {
+        throw new UnsupportedOperationException("decodeFromInt not supported");
+    }
```

Comment:
Could these return `T` instead of `Object` ? And maybe just call them `decode` - the different overloads should be able to disambiguate.